File: selection_sort/selectionSort.js
function findSmaller(arr) {
  let [smaller, smallerIndex] = [arr[0], 0];

  for (let index = 1; index < arr.length; index++) {
    if (arr[index] < smaller) {
      smaller = arr[index];
      smallerIndex = index;
    }
  }
  return smallerIndex;
}

function selectionSort(arr) {
  const sorted = [];
  const copiedArr = [...arr];

  while (copiedArr.length > 0) {
    let smallerPosition = findSmaller(copiedArr);
    let removed = copiedArr.splice(smallerPosition, 1)[0];
    sorted.push(removed);
  }

  return sorted;
}

const testing = [5, 3, 6, 2, 10];
console.log(selectionSort(testing)); // Saída esperada: [2, 3, 5, 6, 10]

File: selection_sort/selection_sort.py
def find_smaller(arr):
    smaller = arr[0]
    smaller_index = 0

    for index in range(1, len(arr)):
        if arr[index] < smaller:
            smaller = arr[index]
            smaller_index = index
    return smaller_index


def selection_sort(arr):
    new_arr = []
    for index in range(len(arr)):
        smaller_position = find_smaller(arr)
        new_arr.append(arr.pop(smaller_position))
    return new_arr


testing_arr = [5, 3, 6, 2, 10]
print(selection_sort(testing_arr))  # Saída esperada: [2, 3, 5, 6, 10]
